marshalers.go

```diff
@@ -43,44 +44,93 @@ func makeMapSyscallOutput(dst any, length int) sysenc.Buffer {
 	return sysenc.SyscallOutput(dst, length)
 }
 
-// marshalPerCPUValue encodes a slice containing one value per
+// appendPerCPUSlice encodes a slice containing one value per
 // possible CPU into a buffer of bytes.
 //
 // Values are initialized to zero if the slice has less elements than CPUs.
-func marshalPerCPUValue(slice any, elemLength int) (sys.Pointer, error) {
+func appendPerCPUSlice(buf []byte, slice any, possibleCPUs, elemLength, alignedElemLength int) ([]byte, error) {
 	sliceType := reflect.TypeOf(slice)
 	if sliceType.Kind() != reflect.Slice {
-		return sys.Pointer{}, errors.New("per-CPU value requires slice")
-	}
-
-	possibleCPUs, err := PossibleCPU()
-	if err != nil {
-		return sys.Pointer{}, err
+		return nil, errors.New("per-CPU value requires slice")
 	}
 
 	sliceValue := reflect.ValueOf(slice)
 	sliceLen := sliceValue.Len()
-	if sliceLen > possibleCPUs {
-		return sys.Pointer{}, fmt.Errorf("per-CPU value exceeds number of CPUs")
+	if sliceLen < possibleCPUs {
```

Comment:
Reading through this again, makes me realise that this is another special case like the one you fixed in https://github.com/cilium/ebpf/commit/bcca828832442d795250ef47840514b81770d281 I'll take a look and see whether I can move the logic so that batch lookups don't inherit this.

Reply:
Yeah I think I got this change the wrong way during testing.

If the slice len is less than possible cpus (orig), we could/can zero pad. Honestly I think it should probably `!=` but there are tests that assume zero pad that I broke with this change so reverting.

Reply:
Yes, we'd still have to zero pad, but maybe I can make it zero pad somewhere else xD